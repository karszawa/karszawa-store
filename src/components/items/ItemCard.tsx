import React from "react";
import styled from "styled-components";
import { ellipsis } from "polished";
import Link from "next/link";
import { A } from "../common/Anchor";
import { LoadingSpan } from "~/styles/loading";

interface ItemCardProps {
  id: string;
  name: string;
  image: { url: string };
  price: number;
}

const ItemCard: React.FC<ItemCardProps> = (props: ItemCardProps) => {
  const { id, image, name, price } = props;

  return (
    <Container>
      <Link href={`/items/${id}`}>
        <A>
          <Img src={image.url} alt={name} />
          {/* loading="lazy" → missing types */}
        </A>
      </Link>
      <Title>
        <Link href={`/items/${id}`}>
          <A>{name}</A>
        </Link>
      </Title>
      <Price>¥ {price}</Price>
    </Container>
  );
};

export default ItemCard;

const Container = styled.div`
  padding: 4px;
  border-radius: 4px;
  box-shadow: 0px 4px 8px 0px rgba(0, 0, 0, 0.15);
  margin: 10px;
`;

const Img = styled.img`
  width: 200px;
  height: 200px;
  object-fit: cover;
  border-radius: 4px;
`;

const Title = styled.h2`
  /* ${ellipsis("auto")} */
  width: auto;
  padding: 12px 12px 0;
  margin: 4px 4px 0;
  border-top: 1.5px solid #eee;
  font-size: 12px;
`;

const Price = styled.span`
  font-weight: bold;
  padding: 12px;
  margin: 4px 4px;
  font-size: 12px;
`;

export const ItemCardLoading: React.FC<{}> = () => (
  <Container>
    <LoadingImg />
    <Title>
      <LoadingSpan>awesome item</LoadingSpan>
    </Title>
    <Price>
      <LoadingSpan>5000 jpy</LoadingSpan>
    </Price>
  </Container>
);

const loadingProps = {
  color: "transparent",
  backgroundColor: "#eee"
};

const LoadingImg = styled.div`
  width: 200px;
  height: 200px;
  border-radius: 4px;
  ${loadingProps}
`;
