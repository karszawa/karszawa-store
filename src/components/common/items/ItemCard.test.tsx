import React from "react";
import { render } from "@testing-library/react";
import "jest-dom/extend-expect";
import ItemCard from "./ItemCard";

test("Render ItemCard", () => {
  const { asFragment } = render(
    <ItemCard
      id="123"
      name="card"
      price={1234}
      image={{ url: "http://placekitten.com/200/300" }}
    />
  );

  expect(asFragment()).toMatchSnapshot();
});
