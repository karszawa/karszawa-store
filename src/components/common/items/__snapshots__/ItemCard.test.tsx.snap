// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ItemCard 1`] = `
<DocumentFragment>
  <div
    class="ItemCard__Container-sc-1q0s1v2-0 gUlvxN"
  >
    <a
      class="Anchor__A-xh1dxk-0 kwIQeY"
    >
      <img
        alt="card"
        class="ItemCard__Img-sc-1q0s1v2-1 dJeIyy"
        src="http://placekitten.com/200/300"
      />
    </a>
    <h2
      class="ItemCard__Title-sc-1q0s1v2-2 htacMv"
    >
      <a
        class="Anchor__A-xh1dxk-0 kwIQeY"
      >
        card
      </a>
    </h2>
    <span
      class="ItemCard__Price-sc-1q0s1v2-3 cXtmZp"
    >
      Â¥ 1234
    </span>
  </div>
</DocumentFragment>
`;
